### PR TITLE
Happy Thoughts API Sandra

### PR DESCRIPTION
**Pull request by sandra-shumanteva-kaevska**
I have done the blue level.

Here is link to my frontend:
https://github.com/sandra-shumanteva-kaevska/project-happy-thoughts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+To create this API I had to work with NodeJs, express, Mongoose and MongoDb. I learned how to create API endpoints, how to create models and how to fetch and create data to MongoDB using Mongoose. Also I used MongoDb Compass. When I got stucked with ideas or soultions I used Google and StackOverflow. While setting the live project I learned to work with mongo cloud and Heroku.
+
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+https://happy-thoughts-sandra.herokuapp.com/
```
Comment (sofiavazs) [inline README.md:12]: 👏🏻

File: server.js [hunk 2]
```diff
@@ -18,11 +18,57 @@ const app = express()
...
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
```
Comment (sofiavazs) [inline server.js:36]: Nice! Mathes the requirements

File: server.js [hunk 3]
```diff
@@ -18,11 +18,57 @@ const app = express()
...
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Happy Thoughts Sandra')
+})
+
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+  res.json(thoughts)
 })
```
Comment (sofiavazs) [inline server.js:46]: 👏🏻

File: server.js [hunk 4]
```diff
@@ -18,11 +18,57 @@ const app = express()
...
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const thoughtId = req.params.thoughtId;
+
+  try {
+    const thoughtLiked = await Thought.updateOne({ _id: thoughtId }, { $inc: { hearts: 1 } });
+    res.json(thoughtLiked);
+  } catch (err) {
+    res.status(400).json({ message: "Thought not found", error: err.errors });
+  }
+})
```
Comment (sofiavazs) [inline server.js:69]: Nice structure and error handling!

**Review by sofiavazs — APPROVED** [review]
Nice job Sandra! Your app works as intended and well done connecting the backend to the frontend! Really easy to read code and nice error handling!

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Please add a link to a frontend to your `README` file.

**Comment by sandra-shumanteva-kaevska** [comment]
> Please add a link to a frontend to your `README` file.

done :)

**Review by KarolCislowski — APPROVED** [review]
Great work 💯

**Pull request closed without merging**